### PR TITLE
vim-patch:0daafaa7d99e

### PR DESCRIPTION
**Pull request by clason**
Update runtime files
https://github.com/vim/vim/commit/0daafaa7d99ef500f76b1b12f5fe8153e2fcaea0

skip vim9script ftplugin
create `userfunc.txt` from Neovim content (skip section 3, needs 9.0.0379)

### CODE REVIEW

File: runtime/doc/userfunc.txt [hunk 1]
```diff
@@ -0,0 +1,428 @@
...
+function, you will get an error message for the missing function.  If you fix
+the autoload script it won't be automatically loaded again.  Either restart
+Vim or manually source the script.
+
+Also note that if you have two script files, and one calls a function in the
+other and vice versa, before the used function is defined, it won't work.
+Avoid using the autoload functionality at the toplevel.
+
+Hint: If you distribute a bunch of scripts you can pack them together with the
+ |vimball| utility.  Also read the user manual |distribute-script|.
```
Comment (seandewar) [inline runtime/doc/userfunc.txt:425]:
```suggestion
|vimball| utility.  Also read the user manual |distribute-script|.
```

**Review by seandewar — COMMENTED** [review]
- `colors/evening.vim` is missing the `g:terminal_ansi_colors` change: https://github.com/vim/vim/commit/0daafaa7d99ef500f76b1b12f5fe8153e2fcaea0#diff-2f89ba28f33c4bdc4c016401580145dbbf3b6b13f392a2e5da215f5372942e71R19
It's N/A for Nvim, but I'd include it anyway to minimise differences.

- `help.txt` needs a link to `userfunc.txt`: https://github.com/vim/vim/commit/0daafaa7d99ef500f76b1b12f5fe8153e2fcaea0#diff-e5cba4644a37135c31c9f067b5d0c5de662134c7d3a6bf5182417eec0844661fR144

- If `:echowindow` is something that we'll port, then we might care to port the `MessageWindow` change in `syntax/vim.vim` at some point.

File: runtime/doc/userfunc.txt [hunk 2]
```diff
@@ -0,0 +1,428 @@
...
+  :endfunction
+<
+When not using "...", the number of arguments in a function call must be at
+least equal to the number of mandatory named arguments.  When using "...", the
+number of arguments may be larger than the total of mandatory and optional
+arguments.
+
+							*local-variables*
+Inside a function local variables can be used.  These will disappear when the
+function returns. Global variables need to be accessed with "g:".
```
[the remark this answers is not in the document]
Comment (seandewar) [reply runtime/doc/userfunc.txt:256]: Actually, this still applies for `v:version` as it's flagged with `VV_COMPAT`, so maybe this should be:

```
Inside functions local variables are accessed without prepending anything.
But you can also prepend "l:" if you like.  This is required for some reserved
names, such as "version".
```

But I'm pretty sure `v:version` is the only case in Nvim..?

File: runtime/doc/builtin.txt [hunk 3]
```diff
@@ -2530,9 +2530,9 @@ function({name} [, {arglist}] [, {dict}])
 		The arguments are passed to the function in front of other
 		arguments, but after any argument from |method|.  Example: >
 			func Callback(arg1, arg2, name)
-			...
+			"...
 			let Partial = function('Callback', ['one', 'two'])
-			...
+			"...
```
Comment (zeertzjq) [inline runtime/doc/builtin.txt:2535]: wrong place

File: runtime/doc/builtin.txt [hunk 4]
```diff
@@ -2563,22 +2563,23 @@ function({name} [, {arglist}] [, {dict}])
...
 			let context = {"name": "example"}
 			let Func = function('Callback', context)
-			...
+			"...
 			call Func()	" will echo: called for example
 <		The use of function() is not needed when there are no extra
-		arguments, these two are equivalent: >
+		arguments, these two are equivalent, if `Callback()` is defined
+		as `context.Callback()`: >
```
Comment (zeertzjq) [inline runtime/doc/builtin.txt:2573]:
```suggestion
		arguments, these two are equivalent, if Callback() is defined
		as context.Callback(): >
```

Comment (clason) [reply runtime/doc/builtin.txt:2573]: This is why we can't have nice things...

File: runtime/doc/userfunc.txt [hunk 5]
```diff
@@ -0,0 +1,433 @@
...
+
+Defining and using functions.
+
+This is introduced in section |41.7| of the user manual.
+
+				      Type |gO| to see the table of contents.
+
+1. Defining a fuction			|define-function|
+2. Calling a fuction			|:call|
+3. Automatically loading functions	|autoload-functions|
```
Comment (seandewar) [inline runtime/doc/userfunc.txt:15]:
```suggestion
```

`gO` makes this redundant

Comment (seandewar) [reply runtime/doc/userfunc.txt]: Looks like the same should be done for `builtin.txt` (and maybe some of the `pi_*` and `usr_*` stuff too, but I wouldn't bother with those here)

**Review by seandewar — APPROVED** [review]

**Pull request merged**